network/transport/grpc/ip_interceptor.go

```diff
@@ -0,0 +1,77 @@
+package grpc
+
+import (
+	"net"
+	"strings"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/peer"
+)
+
+const headerXForwardedFor = "X-Forwarded-For"
+
+// ipInterceptor tries to extract the IP from the X-Forwarded-For header and sets this as the peers address.
+// No address is set if the header is unavailable.
+func ipInterceptor(srv interface{}, serverStream grpc.ServerStream, _ *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+	addr := addrFrom(extractIPFromXFFHeader(serverStream))
```

Comment:
I think this should only be done when TLS is offloaded. Because then, the reverse proxy is expected to set this header. Otherwise, an attacker could set it on the HTTP request, spoofing its IP address.

Reply:
I agree, but this matches the current behavior on the http endpoints. We should change both at the same time.

Reply:
created issue #1416 for this